### PR TITLE
Instantiate environment with asynchronous API

### PR DESCRIPTION
**Pull request by rphmeier**
tiny patch for https://github.com/paritytech/substrate/pull/4630

### CODE REVIEW

File: runtime/kusama/src/lib.rs [hunk 1]
```diff
@@ -77,7 +77,7 @@ pub const VERSION: RuntimeVersion = RuntimeVersion {
 	spec_name: create_runtime_str!("kusama"),
 	impl_name: create_runtime_str!("parity-kusama"),
 	authoring_version: 2,
-	spec_version: 1040,
+	spec_version: 1041,
```
Comment (NikVolf) [inline runtime/kusama/src/lib.rs]: why is this required?

Comment (rphmeier) [reply runtime/kusama/src/lib.rs]: This was part of #769 which I merged in, so @expenses may have more background.

Comment (expenses) [reply runtime/kusama/src/lib.rs]: It might not be needed, CI have me an error before though.

File: validation/src/lib.rs [hunk 2]
```diff
@@ -569,23 +569,24 @@ impl<C, N, P, SC, TxPool, B> consensus::Environment<Block> for ProposerFactory<C
 	// Rust bug: https://github.com/rust-lang/rust/issues/24159
 	sp_api::StateBackendFor<P, Block>: sp_api::StateBackend<HasherFor<Block>> + Send,
 {
+	type CreateProposer = Pin<Box<
```
Comment (mxinden) [inline validation/src/lib.rs]: Is it necessary for this to be `Pin`? In case the future is `Unpin` I would suggest pinning on `poll`ing.

Comment (rphmeier) [reply validation/src/lib.rs]: I guess not - I'll avoid pinning in the follow-up if possible.

Comment (mxinden) [inline validation/src/lib.rs]: The following compiles for me:

```diff
diff --git a/validation/src/lib.rs b/validation/src/lib.rs
index 860c1a6f..51f09700 100644
--- a/validation/src/lib.rs
+++ b/validation/src/lib.rs
@@ -569,9 +569,7 @@ impl<C, N, P, SC, TxPool, B> consensus::Environment<Block> for ProposerFactory<C
        // Rust bug: https://github.com/rust-lang/rust/issues/24159
        sp_api::StateBackendFor<P, Block>: sp_api::StateBackend<HasherFor<Block>> + Send,
 {
-       type CreateProposer = Pin<Box<
-               dyn Future<Output = Result<Self::Proposer, Self::Error>> + Send + Unpin + 'static
-       >>;
+       type CreateProposer = futures::future::Ready<Result<Self::Proposer, Self::Error>>;
        type Proposer = Proposer<P, TxPool, B>;
        type Error = Error;
 
@@ -597,7 +595,7 @@ impl<C, N, P, SC, TxPool, B> consensus::Environment<Block> for ProposerFactory<C
                        backend: self.backend.clone(),
                });
 
-               Box::pin(future::ready(maybe_proposer))
+               future::ready(maybe_proposer)
        }
 }
```

**Comment by mxinden** [comment]
A follow up pull request in regards to my suggestion is fine.
